tests/async.rs

```diff
@@ -93,24 +93,35 @@ where
 
     fn test<C, F>(self, c: C)
     where
-        F: IntoFuture<Item = (), Error = mysql_async::errors::Error>,
-        C: FnOnce(mysql_async::Conn) -> F,
+        F: IntoFuture<Item = (), Error = mysql_async::error::Error> + 'static,
+        <F as futures::IntoFuture>::Future: std::marker::Send,
+        C: FnOnce(mysql_async::Conn) -> F + Send + 'static,
     {
+        let mut runtime = Runtime::new().unwrap();
+
         let listener = net::TcpListener::bind("127.0.0.1:0").unwrap();
         let port = listener.local_addr().unwrap().port();
+
         let jh = thread::spawn(move || {
             let (s, _) = listener.accept().unwrap();
+
             MysqlIntermediary::run_on_tcp(self, s)
         });
 
-        let mut core = Core::new().unwrap();
-        let handle = core.handle();
-        let db =
-            core.run(mysql_async::Conn::new(
-                &format!("mysql://127.0.0.1:{}", port),
-                &handle,
-            )).unwrap();
-        core.run(c(db).into_future()).unwrap();
+        let db_uri = format!("mysql://127.0.0.1:{}", port);
```

Comment:
Since this is only used once, it should probably just be directly in `Pool::new`